pkg/domain/infra/abi/images.go

```diff
@@ -107,6 +108,13 @@ func (ir *ImageEngine) Prune(ctx context.Context, opts entities.ImagePruneOption
 		numPreviouslyRemovedImages = numRemovedImages
 	}
 
+	if opts.BuildCache || opts.All {
```

Comment:
The documentation was not updated and I don't think all is a good fit for this. The all docs say all unused images are removed, this is rather common option because by default prune would only removed untagged images not unused ones with a tag.
So basically everyone runs this will --all I would say and if you know all of the sudden remove the build cahce this feels wrong.

Reply:
Thinking again on this I agree, unless `--build-cache` is specified it should not be cleared, it should either clear `build-cache` on system reset which it already does or when `--build-cache` is explicitly specified. 

@mheon WDYT ? I can open a follow up PR to fix this.